src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/ProcessStartupFilter.java

```diff
@@ -0,0 +1,81 @@
+/* (c) 2015 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wps.executor;
+
+import java.util.Map;
+import java.util.logging.Logger;
+
+import org.geoserver.platform.ExtensionPriority;
+import org.geoserver.wps.process.DelegatingProcessFactory;
+import org.geoserver.wps.process.ProcessFilter;
+import org.geotools.process.Process;
+import org.geotools.process.ProcessException;
+import org.geotools.process.ProcessFactory;
+import org.geotools.util.DelegateProgressListener;
+import org.geotools.util.logging.Logging;
+import org.opengis.feature.type.Name;
+import org.opengis.util.ProgressListener;
+
+/**
+ * A process filter making sure the {@link ProgressListener#started()} method is called upon execution
+ * no matter if the process has inputs or not
+ * 
+ * @author Andrea Aime - GeoSolutions
+ *
+ */
+public class ProcessStartupFilter implements ProcessFilter, ExtensionPriority {
+
+    public class ProcessStartupWrapper implements Process {
+        
+        Process delegate;
+        
+        public ProcessStartupWrapper(Process delegate) {
+            super();
+            this.delegate = delegate;
+        }
+
+        @Override
+        public Map<String, Object> execute(Map<String, Object> input, ProgressListener monitor)
+                throws ProcessException {
+            if(monitor != null) {
+                monitor.started();
+                monitor = new DelegateProgressListener(monitor) {
+                    @Override
+                    public void started() {
+                        // do not pass over, we already called "started"
```

Comment:
Chatting here with Dave and am feeling stupid for not noticing it earlier.

The Process execute method contract asks that the monitor be used (my assumption is started, progress, progress, ... done notifications).

The reflection based process implementation is willing to pass in the monitor for developers to meet this API contract (if the last argument is a monitor). For a no-arguments process (or a process that does not accept a monitor argument) I suspect I have a GeoTools bug where "the monitor" is not being correct notified on execute start and execute completion. 

I *like* your delegate approach (of intercepting the started method) and would like to steal it for a GeoTools fix.